### ISSUE TITLE
Inject deps in objects passed to SMAC

### ISSUE THREAD

**Issue by mlindauer** [automl/SMAC3#187]
Objects such as TAE, intensification and so on have a lot of dependencies (.e.g, intensification needs a runhistory). therefore it is demanding to pass some objects to SMAC.
We should check all objects passed to SMAC whether all dependencies (i.e., refs to other objects) are full filled and fix it if necessary.

**Comment by shukon** [comment]
So basically this will be something like:
```
# initialize intensification or check dependencies
if intensifier is None:
    intensifier = Intensifier(tae_runner=tae_runner,
                                        stats=self.stats,
                                        traj_logger=traj_logger,
                                        rng=rng,
                                        instances=scenario.train_insts,
                                        cutoff=scenario.cutoff,
                                        deterministic=scenario.deterministic,
                                        run_obj_time=scenario.run_obj == "runtime",
                                        instance_specifics=scenario.instance_specific,
                                        minR=scenario.minR,
                                        maxR=scenario.maxR)
elif intensifier.tae_runner != tae_runner:
     intensifier.tae_runner = tae_runner
elif intensifier.stats != self.stats:
     intensifier.stats = self.stats
elif intensifier.traj_logger != traj_logger:
     intensifier.traj_logger = traj_logger
```
for every object, right? Or simply check for the existence? I am not sure - should objects like stats (with scenario being `__scenario`) be re-initialized?

**Comment by shukon** [comment]
On second thought, checking for `None` makes more sense. I will submit a PR.

**Comment by shukon** [comment]
Fixed in  PR #196 (https://github.com/automl/SMAC3/pull/196).
